Bottom: 9211707585a0c931aac7160ab30203227c55130e
Top:    7c19adc0aa0a0e56901b6895e401af1fc2008c17
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-29 15:35:32 +0800

Refresh of ptr_ring-support-zero-length

---

diff --git a/include/linux/ptr_ring.h b/include/linux/ptr_ring.h
index afb2519..42aad79 100644
--- a/include/linux/ptr_ring.h
+++ b/include/linux/ptr_ring.h
@@ -164,7 +164,9 @@ static inline int ptr_ring_produce_bh(struct ptr_ring *r, void *ptr)
  */
 static inline void *__ptr_ring_peek(struct ptr_ring *r)
 {
-	return r->queue[r->consumer];
+	if (likely(r->size))
+		return r->queue[r->consumer];
+	return NULL;
 }
 
 /* Note: callers invoking this in a loop must use a compiler barrier,
